src/common/datetime/format_duration.ts

```diff
@@ -1,23 +1,21 @@
 import { HaDurationData } from "../../components/ha-duration-input";
-import { HomeAssistant } from "../../types";
+import "../../resources/intl-polyfill";
 
 const leftPad = (num: number) => (num < 10 ? `0${num}` : num);
 
-export const formatDuration = (
-  hass: HomeAssistant,
-  duration: HaDurationData
-) => {
+export const formatDuration = (duration: HaDurationData) => {
   const d = duration.days || 0;
   const h = duration.hours || 0;
   const m = duration.minutes || 0;
   const s = duration.seconds || 0;
   const ms = duration.milliseconds || 0;
 
   if (d > 0) {
-    return (
-      hass.localize("ui.duration.day", { count: d }) +
-      ` ${h}:${leftPad(m)}:${leftPad(s)}`
-    );
+    return `${Intl.NumberFormat("en", {
```

Comment:
Can we pass the locale of the user?

Reply:
And move it to format date/format time

Reply:
I think you may have reviewed the wrong commit. User locale is already passed in.